### ISSUE TITLE
nematic order parameter crashes

### ISSUE THREAD

**Issue by jglaser** [glotzerlab/freud#264]
**Describe the bug**

Using the nematic order parameter results in a crash.

**To Reproduce**

```
import signac
import numpy as np
import sys

from freud import order
from glotzformats.reader import GSDHOOMDFileReader

gsd_reader = GSDHOOMDFileReader()

def process_job(job):
    with job:
        try:
            with open('out.gsd','rb') as gsdfile:
                traj = gsd_reader.read(gsdfile)
#                nematic = order.NematicOrderParameter(u=(0,0,1))
                nematic = order.NematicOrderParameter(u=(1,0,0))

                nematic.compute(traj[-1].orientations)
                sp = job.statepoint()
                eta_dep = sp['eta_dep']
                r_dep = sp['r_dep']
                phi = sp['phi']
                op = nematic.get_nematic_order_parameter()
                return op
        except Exception:
            return 0

project = signac.get_project()
jobs = list(project.find_jobs())

from multiprocessing import Pool
pool = Pool()
ops = pool.map(process_job,jobs)
```

**Error output**
If applicable, add terminal output or screenshots to help explain your problem.
```
(myenv) [glaser@login5.summit janus_n10_fine]$ python nematic_perp.py 2>&1 | more
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
thread_monitor Resource temporarily unavailable in pthread_create
*** Error in `python': free(): corrupted unsorted chunks: 0x0000000130d393b0 ***
thread_monitor Resource temporarily unavailable in pthread_create
...
```

**System configuration (please complete the following information):**
 - OS: ppc linux
 - Version of Python: anaconda python 3.7.1
 - Version of freud : 1c5af70f7dfaf104c124b192ed579ccd33f3da01

**Additional context**
Please let me know if I should share the .gsd files with you.

**Comment by bdice** [comment]
@jglaser and I spent some time debugging this in person. There are some memory management issues that only seem to appear on POWER processors. We tried to run with an address sanitizer and started work on another problem related to AABBQuery and a TBB variable being double-freed. I do not know how to proceed further at this time.

**Comment by jglaser** [comment]
> @jglaser and I spent some time debugging this in person. There are some memory management issues that only seem to appear on POWER processors. We tried to run with an address sanitizer and started work on another problem related to AABBQuery and a TBB variable being double-freed. I do not know how to proceed further at this time.

This is not a short term solution, but going forward it looked like this and similar issues may be linked to the fragile cython glue between python and C++. We could open an issue with the intent of eliminating that layer in favor of pybind11.

**Comment by bdice** [comment]
@jglaser Can you try the branch `fix/test-precision` and see if you're still seeing problems after we switched to the `SelfAdjointEigenSolver`? The tests pass now, so I hope this issue is resolved.

**Comment by bdice** [comment]
I think this was solved in #254 - if problems continue, please re-open this issue.
